### PR TITLE
Add active block variation information in Block Parent Selector

### LINKED ISSUE #30665: Block parent selector should acknowledge semantic template part data
If you select a block inside a semantic template part like a header, the parent selector that appears adjacent to the toolbar should display the appropriate icon, and the tooltip should include the appropriate name rather than the generic "Template Part" label.

Here's a screenshot of the current experience:

Screenshot 2021-04-09 at 14 17 30

In this case the icon that I am hovering (the blue one) should be the [header](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/header.js) icon from the core package. The tooltip should read "Select Header".

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves: https://github.com/WordPress/gutenberg/issues/30665

This PR extracts active block variation's information (if any) in Block parent selector component.

## Testing instructions
This can be tested now in trunk with `Template Part` blocks.
If you select a block inside a semantic template part like a `header`, the parent selector that appears adjacent to the toolbar will now display the appropriate icon, and the tooltip should include the appropriate name rather than the generic "Template Part" label.

### CODE REVIEW

File: packages/block-editor/src/components/block-parent-selector/index.js [hunk 1]
```diff
@@ -24,36 +25,34 @@ export default function BlockParentSelector() {
...
-			firstParentClientId: _firstParentClientId,
-			shouldHide: ! hasBlockSupport(
-				_parentBlockType,
-				'__experimentalParentSelector',
-				true
-			),
-			hasReducedUI: settings.hasReducedUI,
-		};
-	}, [] );
+	const { firstParentClientId, shouldHide, hasReducedUI } = useSelect(
```
Comment (ntsekouras) [inline packages/block-editor/src/components/block-parent-selector/index.js:28]: This diff is the result of just removing the `parentBlockType` prop, which is not used anymore in favor of `useBlockDisplayInformation`.

Comment (gziolo) [reply packages/block-editor/src/components/block-parent-selector/index.js:28]: https://github.com/WordPress/gutenberg/pull/30731/files?diff=split&w=1 is always helpful

Screen Shot 2021-04-12 at 12 04 14

**Review by gziolo — APPROVED** [review]
Nice one 👍🏻

**Review by jameskoster — APPROVED** [review]
Yay!

**Pull request merged**